qualtran/_infra/registers.py

```diff
@@ -99,6 +100,10 @@ def total_bits(self) -> int:
 
         This is the product of bitsize and each of the dimensions in `shape`.
         """
+        return self._total_bits
+
+    @cached_property
```

Comment:
just use cached method on the original method?

Reply:
`functools` doesn't seem to have a `cached_method` and I didn't want to add another dependency. using just `cache` would require computing hash of the class itself which would again be slow.